tesseract_environment/src/environment.cpp

```diff
@@ -917,6 +942,14 @@ void Environment::environmentChanged()
     group_joint_names_cache_.clear();
   }
 
+  // Call the event callbacks
+  if (!event_cb_.empty())
+  {
+    CommandAppliedEvent event(commands_, revision_);
+    for (const auto& cb : event_cb_)
+      cb(event);
```

Comment:
So all callbacks get called with this even regardless of event type? It seems you could also do this by storing them in a map by event type and only calling the ones for this event type.

Reply:
I was following a similar pattern to how [Qt event filters](https://doc.qt.io/qt-5/eventsandfilters.html#event-filters) work. They then just have a switch statement in there code to handle the events they care about.

Reply:
I think this list should be small and have little impact but may need to revisit if that turns out to not be the case.